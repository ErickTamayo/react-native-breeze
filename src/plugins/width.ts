import { PluginFunction, PluginPattern } from "./types";

export type PluginGroups = {
  key: string;
};

export const pattern: PluginPattern = ({ keys }) => {
  return new RegExp(`^w-(?<key>${keys("width")})$`);
};

export const plugin: PluginFunction<PluginGroups> = ({ groups, theme }) => {
  const { key } = groups;
  const value = theme(["width", key]);
  return { width: value };
};
