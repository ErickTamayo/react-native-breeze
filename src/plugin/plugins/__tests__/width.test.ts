import {
  generateInput,
  shouldEvaluateTheCorrectPatternTest,
  shouldMatchOutputSnapshot,
} from "../../utils/tests";
import { pattern, plugin } from "../width";

describe("width", () => {
  const shouldMatch = generateInput("w", ["1", "2", "3", "4", "5", "6", "8"]);

  shouldEvaluateTheCorrectPatternTest(pattern, {
    shouldMatch,
    shouldNotMatch: ["unknown", "w-unknown"],
  });

  shouldMatchOutputSnapshot(pattern, plugin, shouldMatch);
});
