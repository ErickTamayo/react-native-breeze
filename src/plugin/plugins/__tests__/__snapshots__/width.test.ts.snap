// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`width Should parse (w-1) correctly 1`] = `
Object {
  "width": 4,
}
`;

exports[`width Should parse (w-2) correctly 1`] = `
Object {
  "width": 8,
}
`;

exports[`width Should parse (w-3) correctly 1`] = `
Object {
  "width": 12,
}
`;

exports[`width Should parse (w-4) correctly 1`] = `
Object {
  "width": 16,
}
`;

exports[`width Should parse (w-5) correctly 1`] = `
Object {
  "width": 20,
}
`;

exports[`width Should parse (w-6) correctly 1`] = `
Object {
  "width": 24,
}
`;

exports[`width Should parse (w-8) correctly 1`] = `
Object {
  "width": 32,
}
`;
